store/etcd/etcd.go

```diff
@@ -735,31 +766,71 @@ func (l *etcdLock) holdLock(key string, lockHeld chan struct{}, stopLocking <-ch
 	for {
 		select {
 		case <-update.C:
+			fmt.Printf("=====> %d: etcd.holdLock(): got update, setting %s to %s with prevIndex %d\n", getGID(), key, l.value, setOpts.PrevIndex)
 			setOpts.PrevIndex = l.last.Node.ModifiedIndex
 			l.last, err = l.client.Set(context.Background(), key, l.value, setOpts)
 			if err != nil {
+				fmt.Printf("=====> %d: etcd.holdLock(): got error %s on setting %s to %s\n", getGID(), err, key, l.value)
 				return
 			}
 
-		case <-stopLocking:
+		case msg := <-stopLocking:
+			fmt.Printf("=====> %d: etcd.holdLock(): got stopLocking: %p\n", getGID(), &msg)
 			return
 		}
 	}
 }
 
 // WaitLock simply waits for the key to be available for creation
```

Comment:
If this returns, the caller just knows that the lock is available, right? It doesn't actually take the lock? Isn't there a chance that the lock will be taken between this returning and the caller finally trying to take the lock?

Reply:
Yes there is but that part is prevented by the CAS mechanism.